src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -485,10 +499,18 @@ def _init_optimizers(self) -> Tuple[Optimizer, Optional[LRSchedulerConfig], Opti
 
     @property
     def zero_stage_3(self) -> bool:
-        return self.config.get("zero_optimization") and self.config.get("zero_optimization").get("stage") == 3
+        assert isinstance(self.config, dict)
+        zero_optimization = self.config.get("zero_optimization")
+        if zero_optimization is None:
+            raise MisconfigurationException(
+                "You must set `zero_optimization` in the DeepSpeed config."
+                " See: https://pytorch-lightning.readthedocs.io/en/stable/advanced/model_parallel.html#deepspeed"
+            )
+        return bool(self.config.get("zero_optimization")) and zero_optimization.get("stage") == 3
```

Comment:
should we raise Exceptions inside properties? Ideally, it should just check and return False if not found.
```suggestion
        assert isinstance(self.config, dict)
        zero_optimization = self.config.get("zero_optimization")
        return zero_optimization and zero_optimization.get("stage") == 3
```

Reply:
We'd probably not. Thanks!